run-nondex.sh

```diff
@@ -25,9 +25,17 @@ sha=$(git rev-parse HEAD)
 # Run NonDex, NUMROUNDS rounds
 timeout 3600s mvn edu.illinois:nondex-maven-plugin:2.1.1:nondex -DnondexRuns=${NUMROUNDS}
 
+# Run tests
+timeout 3600s mvn test > test_output.txt 2>&1
+
 # Grab all the detected tests
 if [[ "$(find -name failures | wc -l)" != "0" ]]; then 
     for t in $(cat $(find -name failures) | sort -u); do
-	echo ${slug},${sha},${t} >> ${resultsfile}
+        # If the fully-qualified test name is also present in the test_output file then this test always fails 
```

Comment:
Why would a test that exist in test_output.txt mean that it always fails?

The test_output.txt is just the log of running mvn test. This log may 
(1) also output the names of tests that passed
(2) not output names of tests that fail depending on the version of Surefire.

Instead you may just process the Surefire reports directly to see if the tests passed or not. See the following for some examples of how to parse Surefire XML reports (target/surefire-reports).
https://github.com/winglam/azure-tools/blob/master/python-scripts/parse_surefire_report.py
https://github.com/winglam/azure-tools/blob/master/rounds.sh#L30

Reply:
In fact, just running mvn test once may not tell you if the test is deterministically failing or if the test is just flaky in that one run of mvn test. 

We would need to either run mvn test 3 times and ensure that the test fails all 3 times OR directly get the list of failing tests from NonDex. 

The latter option is likely best. I can ask the NonDex authors for how to do this latter option but please look inside the .nondex files and the NonDex documentation (e.g., GitHub) to see if you can figure out how to get this information from running NonDex (and not even need to run mvn test at all).

Reply:
I spoke with NonDex authors. One option here is for you to parse the Surefire reports stored under .nondex/. If a test failed in all of the surefire reports, then it should not be reported as flaky.

Another way of finding the tests that failed deterministically from the flaky tests is just better parsing of the log from running NonDex. **In fact, this way may be the simplest and best way to separate NonDex flaky tests from deterministically failing tests.**

This is an example log:
```
[INFO] The following tests failed in the clean run:
[WARNING] ta4jexamples.analysis.CashFlowToChartTest#test
[WARNING] ta4jexamples.analysis.BuyAndSellSignalsToChartTest#test
[WARNING] ta4jexamples.indicators.IndicatorsToChartTest#test
[WARNING] ta4jexamples.indicators.CandlestickChartTest#test
[INFO] ####################
[INFO] Across all seeds:
[INFO] Test results can be found at:
```
If you had tests that (1) passed without shuffling and (2) failed with shuffling, those would likely be flaky tests, and they would appear in the line(s) below "Across all seeds" and before "Test results can be found at". While tests that appear under "The following tests failed in the clean run:" are just deterministically failing tests.

Reply:
I have updated the run-nondex.sh file to get the flaky tests by parsing the log of NonDex run. Here, the tests that appeared after 'Across all seeds' and before 'Test results can be found at' are taken as flaky tests.